src/server/internalServer.cpp

```diff
@@ -192,9 +192,10 @@ SearchInfo::SearchInfo(const std::string& pattern, GeoQuery geoQuery)
     geoQuery(geoQuery)
 {}
 
-SearchInfo::SearchInfo(const RequestContext& request)
+SearchInfo::SearchInfo(const RequestContext& request, const std::set<std::string>& bookIds)
```

Comment:
This constructor has became ugly. Parsing of the request better be consolidated in one place. The constructor must be

```c++
SearchInfo(const std::string& pattern, const GeoQuery&, const BookIdSet&)
```

and there must be a method

```c++
SearchInfo InternalServer::getSearchInfo(const RequestContext& ) const;
```

Reply:
Done